src/fullnode.rs

```diff
@@ -24,30 +24,42 @@ pub struct FullNode {
     pub thread_hdls: Vec<JoinHandle<()>>,
 }
 
+pub enum LedgerFile {
+    NoFile,
+    StdIn,
+    StdOut,
+    Sink,
+    Path(String),
+}
+
 impl FullNode {
     pub fn new(
         mut node: TestNode,
         leader: bool,
-        infile: Option<String>,
+        infile: LedgerFile,
         network_entry_for_validator: Option<SocketAddr>,
-        outfile_for_leader: Option<String>,
+        outfile_for_leader: LedgerFile,
         exit: Arc<AtomicBool>,
     ) -> FullNode {
         info!("creating bank...");
         let bank = Bank::default();
-        let entry_height = if let Some(path) = infile {
-            let f = File::open(path).unwrap();
-            let mut r = BufReader::new(f);
-            let entries =
-                entry_writer::read_entries(&mut r).map(|e| e.expect("failed to parse entry"));
-            info!("processing ledger...");
-            bank.process_ledger(entries).expect("process_ledger")
-        } else {
-            let mut r = BufReader::new(stdin());
-            let entries =
-                entry_writer::read_entries(&mut r).map(|e| e.expect("failed to parse entry"));
-            info!("processing ledger...");
-            bank.process_ledger(entries).expect("process_ledger")
+        let entry_height = match infile {
+            LedgerFile::Path(path) => {
+                let f = File::open(path).unwrap();
+                let mut r = BufReader::new(f);
+                let entries =
+                    entry_writer::read_entries(&mut r).map(|e| e.expect("failed to parse entry"));
```

Comment:
This will probably cause the full ledger to be pulled into memory.

Reply:
there is no collect, i thought map just wraps it with an iter

Reply:
I'll have to investigate. I don't think `map f . map g => map (f . g)` is implemented, and I'm not sure if mapping on an iterator avoids that.